### DISCUSSION TITLE
How to use the Tasmota native Counter in Berry?

### DISCUSSION THREAD

**Discussion by mrxllone** [arendst/Tasmota#19306] [Q&A] [answered]
Hi there!

I am bulding a project which uses a water flow sensor, sending out pulses. I am using the Tasmota component "Counter" to capture and count the pulses, which works perfectly fine.
Now, using Berry I want to convert the number of pulses to the volume flowing though it, by multiplication with a constant, along with doing further functions.

For that, I currently get the counter value every 250ms using `self.p_last_sec = tasmota.cmd("Counter1")["Counter1"]` in Berry, and reset it using `tasmota.cmd("Counter1 0")`. Although this works just fine, it is "spamming" the MQTT topic "RESULT", obviously as a result of the Tasmota command `Counter1` being called 4 times a second.

I wonder if there is a better way to read and set the counter value from within Berry, similar to what is possible with `tasmota.get_power()` or `tasmota.set_power(int, bool)`. I did not find such a function.

**Comment by sfromis** [comment, marked as answer]
You can reduce the console logging like: `tasmota.cmd("Counter1", true)["Counter1"]` - this does still not reduce the MQTT logging. Note that using `tasmota.set_power(int, bool)` still results in console logging when you use that function.

For what you describe, I'd just not worry about resetting the counter all the time, but save the previous value and subtract. Not so much due to the logging, but to make sure not to lose any ticks.

Likewise, I'd also calculate the time between each call, to deal with the risk of other activities resulting in the interval not being rock solid 250 ms.

**Reply by mrxllone** [reply]
Thanks for the hints! It really is useless to reset the counter all the time, that alone cuts the number of messages in half. I'll also look into doing the time interval calculation as well.

And for now, I'll settle with the fact that there's no option to suppress the MQTT messages.
